### ISSUE TITLE
Record versioning/edit history

### ISSUE THREAD

**Issue by knirirr** [FAIRsharing/FAIRsharing-API#8]
Once #5 and #6 have been addressed then Papertrail should be integrated (https://github.com/paper-trail-gem/paper_trail). Eventually it should be integrated with the admin interface, though this is probably outside the scope of the initial ticket (#7).

**Comment by knirirr** [comment]
I've added the following:

https://github.com/paper-trail-gem/paper_trail
https://github.com/westonganger/paper_trail-association_tracking

Some models could probably be excluded (e.g. tags) but it's not clear yet exactly what will be required and so I have enabled history for all models and controllers. But, note the caveats on the second link, for example 

> 
This plugin currently can restore three types of associations: Has-One, Has-Many, and Has-Many-Through. 

I.e. has_and_belongs_to_many relationships will not be covered, which for fairsharing_records includes:

```
 has_and_belongs_to_many :subjects, -> { distinct }
 has_and_belongs_to_many :domains, -> { distinct }
 has_and_belongs_to_many :taxonomies, -> { distinct } # TODO: Should be joined to "not applicable" by default
 has_and_belongs_to_many :user_defined_tags, -> { distinct }, join_table: 'fsrs_to_user_defined_tags'
 has_and_belongs_to_many :countries, -> { distinct }
```

This ticket hasn't been closed yet as some further thought and checking may well be required.
